File: jambeth/jAmbeth.Persistence/src/de/osthus/ambeth/database/ResultingDatabaseCallback.java
package de.osthus.ambeth.database;

import de.osthus.ambeth.collections.ILinkedMap;
import de.osthus.ambeth.persistence.IDatabase;

public interface ResultingDatabaseCallback<R>
{
	R callback(ILinkedMap<Object, IDatabase> persistenceUnitToDatabaseMap) throws Exception;
}

File: ambeth/Ambeth.Persistence.API/ambeth/database/ResultingDatabaseCallback.cs
using De.Osthus.Ambeth.Collections;
using De.Osthus.Ambeth.Persistence;
using System;

namespace De.Osthus.Ambeth.Database
{
    public delegate R ResultingDatabaseCallback<R>(ILinkedMap<Object, IDatabase> persistenceUnitToDatabaseMap);
}